tools-v2/pkg/cli/command/curvebs/update/scan_state/scan_state.go

```diff
@@ -0,0 +1,151 @@
+/*
+ *  Copyright (c) 2023 NetEase Inc.
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License");
+ *  you may not use this file except in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ */
+
+/*
+* Project: curvecli
+* Created Date: 2023-05-08
+* Author: montaguelhz
+ */
+
+package scan_state
+
+import (
+	"context"
+	"fmt"
+
+	cmderror "github.com/opencurve/curve/tools-v2/internal/error"
+	cobrautil "github.com/opencurve/curve/tools-v2/internal/utils"
+	basecmd "github.com/opencurve/curve/tools-v2/pkg/cli/command"
+	"github.com/opencurve/curve/tools-v2/pkg/config"
+	"github.com/opencurve/curve/tools-v2/pkg/output"
+	"github.com/opencurve/curve/tools-v2/proto/proto/topology"
+	"github.com/opencurve/curve/tools-v2/proto/proto/topology/statuscode"
+	"github.com/spf13/cobra"
+	"google.golang.org/grpc"
+)
+
+type UpdateScanStateRpc struct {
+	Info    *basecmd.Rpc
+	Request *topology.SetLogicalPoolScanStateRequest
+	Client  topology.TopologyServiceClient
+}
+
+var _ basecmd.RpcFunc = (*UpdateScanStateRpc)(nil) // check interface
+
+func (sRpc *UpdateScanStateRpc) NewRpcClient(cc grpc.ClientConnInterface) {
+	sRpc.Client = topology.NewTopologyServiceClient(cc)
+}
+
+func (sRpc *UpdateScanStateRpc) Stub_Func(ctx context.Context) (interface{}, error) {
+	return sRpc.Client.SetLogicalPoolScanState(ctx, sRpc.Request)
+}
+
+type ScanStateCommand struct {
+	basecmd.FinalCurveCmd
+
+	Rpc           *UpdateScanStateRpc
+	Response      *topology.SetLogicalPoolScanStateResponse
+	LogicalPoolID uint32
+	Scan          bool
+}
+
+var _ basecmd.FinalCurveCmdFunc = (*ScanStateCommand)(nil) // check interface
+
+const (
+	scanStateExample = `$ curve bs update scan-state --logicalpoolid 1 [--scan=true/false]`
+)
+
+func NewScanStateCommand() *cobra.Command {
+	return NewUpdateScanStateCommand().Cmd
+}
+
+func (sCmd *ScanStateCommand) AddFlags() {
+	config.AddBsMdsFlagOption(sCmd.Cmd)
+	config.AddRpcRetryTimesFlag(sCmd.Cmd)
+	config.AddRpcTimeoutFlag(sCmd.Cmd)
+
+	config.AddBSLogicalPoolIdRequiredFlag(sCmd.Cmd)
+	config.AddBsScanOptionFlag(sCmd.Cmd)
```

Comment:
Flag scan is required, will it be better?

Reply:
In my opinion, from the user's point of view, it is more intuitive for the scan flag to default to true. At the same time, reducing parameters can reduce the user's memory burden.